src/librbd/api/Image.cc

```diff
@@ -302,6 +310,11 @@ int Image<I>::deep_copy(I *src, librados::IoCtx& dest_md_ctx,
     return r;
   }
 
+  if (dst_size != src_size) {
+    NoOpProgressContext no_op;
+    r = dest->operations->resize(src_size, true, no_op);
```

Comment:
Can we just adjust the check in ```SetHeadRequest::send_remove_parent``` to remove the parent if the overlap doesn't match?

Reply:
Probably call `cls::set_size` instead

Reply:
Oh, we already calls set_size as the first step. So it should update the overlap. It looks like we just don't need to call remove/set parent when overlap does not match. I will try this.